src/components/MapView.js

```diff
@@ -0,0 +1,427 @@
+        const districtPadded =
+          zeros.substring(0, zeros.length - districtString.length) +
+          districtString;
+        // highlight district
+        const geoID = `${stateFIPS}${districtPadded}`;
+        const selectObj = {
+          district: districtPadded,
+          geoID,
+          state: nextProps.selectedState,
+        };
+        this.districtSelect(selectObj);
+      }
+      const stateBB = bboxes[bbname];
+      return this.focusMap(stateBB);
+    }
+    this.map.metadata.level = 'state';
+    this.setStateStyleMask();
+
+    return this.map.fitBounds([[-128.8, 23.6], [-65.4, 50.2]]);
+  }
+
+  setDistrictStyle() {
+    const { items } = this.props;
+    const lowNumbers = ['any'];
+    const medNumbers = ['any'];
+    const highNumbers = ['any'];
+    Object.keys(items).forEach((state) => {
+      let count = 0;
+      Object.keys(items[state]).forEach((district) => {
+        const districtId = district;
+        const fipsId = fips[state];
+        const geoid = fipsId + districtId;
+        count += filter((items[state][district]), 'pledged').length;
+        if (count >= 3) {
+          highNumbers.push(['==', 'GEOID', geoid]);
+        } else if (count >= 2) {
+          medNumbers.push(['==', 'GEOID', geoid]);
+        } else if (count > 0 && count < 2) {
+          lowNumbers.push(['==', 'GEOID', geoid]);
+        }
+      });
+    });
+    this.toggleFilters('high_number', highNumbers);
+    this.toggleFilters('med_number', medNumbers);
+    this.toggleFilters('low_number', lowNumbers);
+  }
+
+  setStateStyleMask(state) {
+    if (state) {
+      const filterSetting = ['!=', 'ref', state];
+      this.toggleFilters('state-mask', filterSetting);
+    }
+    else {
+      this.map.setLayoutProperty('state-mask', 'visibility', 'none');
+    }
+  }
+
+
+  setStateStyle() {
+    const { items } = this.props;
+    const lowNumbers = ['in', 'ABR'];
+    const medNumbers = ['in', 'ABR'];
+    const highNumbers = ['in', 'ABR'];
+    Object.keys(items).forEach((state) => {
+      let count = 0;
+      Object.keys(items[state]).forEach((district) => {
+        count += filter((items[state][district]), 'pledged').length;
+      });
+      if (count >= 10) {
+        highNumbers.push(state);
+      } else if (count >= 4) {
+        medNumbers.push(state);
+      } else if (count > 0 && count < 4) {
+        lowNumbers.push(state);
+      }
+    });
+
+    this.toggleFilters('high_number', highNumbers);
+    this.toggleFilters('med_number', medNumbers);
+    this.toggleFilters('low_number', lowNumbers);
+  }
+
+  addClickListener() {
+    const {
+      searchByDistrict,
+    } = this.props;
+    const { map } = this;
+
+    map.on('click', (e) => {
+      const features = map.queryRenderedFeatures(
+        e.point,
+        {
+          layers: ['district_interactive'],
+        },
+      );
+      const feature = {};
+      if (features.length > 0) {
+        feature.state = features[0].properties.ABR;
```

Comment:
So we have the feature(s) here, as well as the container layer/feature.  We should be able to use it to automatically set zoom instead of using the bboxes.